src/nvim/api/mark_extended.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Same function but names are not limited to one char
+ * There is also no ex_command, just viml functions
+ */
+
+#include "nvim/vim.h"
+#include "nvim/globals.h"      // FOR_ALL_BUFFERS
+#include "nvim/mark.h"         // SET_FMARK
+#include "nvim/memory.h"
+#include "nvim/map.h"          // pmap ...
+#include "nvim/lib/kbtree.h"   // kbitr ...
+#include "nvim/api/mark_extended.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "api/extmark.c.generated.h"
+#endif
+
+#define FOR_ALL_EXTMARKS(buf) \
+  kbitr_t itr; \
+  ExtendedMark *extmark; \
+  for (;kb_itr_valid(&itr); kb_itr_next(ExtendedMark, buf->b_extmarks_tree, &itr)){
+
+/* Create or update an extmark, */
+int extmark_set(buf_T *buf, cstr_t *name, pos_T *pos)
```

Comment:
I also have an impression that what I already said was not the only issue with map interface, but cannot remember anything else. Perhaps one of the issues was that public API is too complex: when you declare a new hash using `KHASH_…` you get nice names like `kh_init_foo`. When you declare a new hash using map interface you need to use `map_new(Type1, Type2)`. Worse, you need to typedef Type1 and Type2 in some cases I would not typedef and this is absolutely not needed when using khash.